code/modules/admin/ipintel.dm

```diff
@@ -0,0 +1,224 @@
+/datum/ipintel
+	var/ip
+	var/intel = 0
+	var/cache = FALSE
+	var/cacheminutesago = 0
+	var/cachedate = ""
+	var/cacherealtime = 0
+
+/datum/ipintel/New()
+	cachedate = SQLtime()
+	cacherealtime = world.realtime
+
+/datum/ipintel/proc/is_valid()
+	. = FALSE
+	if(intel < 0)
+		return
+	if(intel <= config.ipintel_rating_bad)
+		if(world.realtime < cacherealtime + (config.ipintel_save_good * 60 * 60 * 10))
```

Comment:
Reviewer's note:
This proc seems to be about when to update a cached ipintel entry? And this seems to be in terms of hours, right?
Wouldn't it be better to use the `HOURS` macro, which does this `60 * 60 * 10` multiplication, but makes the intent very explicit?

Reply:
Changed, please check that the new format is correct use of the macro.